packages/@aws-cdk/aws-servicecatalog/README.md

```diff
@@ -184,6 +185,78 @@ const product = new servicecatalog.CloudFormationProduct(this, 'Product', {
 });
 ```
 
+### Creating a Product from a stack with immutable versions
+
+The default behavior of Service Catalog is to overwrite each product version upon deployment.
+This applies to Product Stacks as well, where only the latest changes to your Product Stack will 
+be deployed.
+We would need to create a separate Product Stack to store that version of Product Stack in code.
+If instead you want to never overwrite existing versions, but only add new versions, you can use the
+`RetentionStrategy.RETAIN` strategy for your deployment.
+
+```ts
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+
+class S3BucketProduct extends servicecatalog.ProductStack {
+  constructor(scope: cdk.Construct, id: string) {
+    super(scope, id);
+
+    new s3.Bucket(this, 'BucketProduct');
+  }
+}
+
+const product = new servicecatalog.CloudFormationProduct(this, 'Product', {
+  productName: "My Product",
+  owner: "Product Owner",
+  productVersions: [
+    {
+      productVersionName: "v1",
+      cloudFormationTemplate: servicecatalog.CloudFormationTemplate.fromProductStack(new S3BucketProduct(this, 'S3BucketProduct'), servicecatalog.RetentionStrategy.RETAIN),
+    },
+  ],
+});
+```
+
+Using the `RetentionStrategy.RETAIN` strategy all deployed templates for the ProductStack will be written to disk,
+so that they will still be available in the future as the definition of the `ProductStack` subclass changes over time.
+**It is very important** that you commit these old versions to source control as these versions 
+determine whether a version has already been deployed and can be used also be deployed themselves 
+using `fromProductStackSnapshot`.
+
+After using the `RetentionStrategy.RETAIN` strategy to deploy version `v1` of your `ProductStack`, we 
+make changes to the `ProductStack` and update it to `v2`.
+We also want our `v1` version to still be deployed, so we reference it using `fromProductStackSnapshot`
+and passing the corresponding `ProductStack` id.
+
+```ts
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+
+class S3BucketProduct extends servicecatalog.ProductStack {
+  constructor(scope: cdk.Construct, id: string) {
+    super(scope, id);
+
+    new s3.Bucket(this, 'BucketProductV2');
+  }
+}
+
+const product = new servicecatalog.CloudFormationProduct(this, 'MyFirstProduct', {
+  productName: "My Product",
+  owner: "Product Owner",
+  productVersions: [
+    {
+      productVersionName: "v2",
+      cloudFormationTemplate: servicecatalog.CloudFormationTemplate.fromProductStack(new S3BucketProduct(this, 'S3BucketProduct'), servicecatalog.RetentionStrategy.RETAIN),
```

Comment:
Let's rename `RetentionStrategy` so that it's very clear that what you put here relates to `fromProductStackSnapshot`:

```suggestion
      cloudFormationTemplate: servicecatalog.CloudFormationTemplate.fromProductStack(new S3BucketProduct(this, 'S3BucketProduct'), servicecatalog.Snapshot.CREATE_SNAPSHOT),
```

I think we have the following options:

- `Snapshot.NONE`: default
- `Snapshot.CREATE_SNAPSHOT`: creates and overwrites snapshot
- `Snapshot.LOCK_SNAPSHOT`: creates and verifies snapshot

People can write code to switch between the 2 snapshot flavors depending whether they're doing a DEV or CI build.